wiki/markdown/serviceUsage.md

```diff
@@ -0,0 +1,443 @@
+
+Start the application from Tomcat.
+
+### Jasypt configurations
+
+Jasypt Spring Boot provides Encryption support for property sources(passwords, secret info ..etc) in Spring Boot Applications. To support this functionality in our application we need to add the following property in property file
+
+```
+jasypt.encryptor.password: The key value which was used while encrypting the original password
+```
+
+The above encryptor password must be same for both encryption and decryption of the original password(ldap,rabbitmq...etc)
+The encryptor password will be used by jasypt-spring-boot library in application to decrypt the encrypted password at runtime.
+
+### How to encrypt the password:
+
+```
+1) Download the jasypt jar file from any of the below locations
+Link to [Jasypt](http://www.jasypt.org/download.html) (or) [Link to Maven](https://mvnrepository.com/artifact/org.jasypt/jasypt/1.9.2)
+
+2) Execute the below command to generate Jasypt encrypted password
+    java -cp jasypt-1.9.2.jar org.jasypt.intf.cli.JasyptPBEStringEncryptionCLI input="any password" password=any intermediate key
+
+input   : any password which we want to encrypt(Ex: rabbitmq password, ldap user password, etc...)
+password: A Jasypt key used to encrypt the above input( The Jasypt key can be anything, but make sure same key to be used for decryption)
+
+Example:
+    Run the below command in Command line
+
+    java -cp jasypt-1.9.2.jar org.jasypt.intf.cli.JasyptPBEStringEncryptionCLI input="dummyPassword" password=dummy
+
+output:
+    ----ENVIRONMENT-----------------
+
+    Runtime: Oracle Corporation Java HotSpot(TM) 64-Bit Server VM 25.144-b01
+
+    ----ARGUMENTS-------------------
+
+    algorithm: PBEWithMD5AndDES
+    input: dummyPassword
+    password: dummy
+
+    ----OUTPUT----------------------
+
+    euJcvto7NtCDiWT7BKFW0A==
+```
+
+Use the above encrypted password in your property file like this **ENC(encrypted password)**
+
+`Ex:rabbitmq.password: ENC(euJcvto7NtCDiWT7BKFW0A ==)`
+
+**Note: REMReM will work without jasypt encryption also but if you have encrypted any text using jasypt library then jasypt.encryptor.password property should be present in property file.**
+
+### Exchange configurations
+
+These parameters are related to RabbitMQ Server, which will be used for publishing events:
+
+```
+<protocol>.rabbitmq.host:          <host name, eg: localhost>
+<protocol>.rabbitmq.port:          <port, eg: 5672>
+<protocol>.rabbitmq.username:      <username, default for RabbitMQ Server: guest>
+<protocol>.rabbitmq.password:      <password, default for RabbitMQ Server: guest>
+<protocol>.rabbitmq.tls:           <tls version, is optional>
+<protocol>.rabbitmq.exchangeName:  <exchange name, exchange should be already created on RabbitMQ Server>
+<protocol>.rabbitmq.createExchangeIfNotExisting:  <create Exchange if not present on RabbitMQ Server>
+<protocol>.rabbitmq.domainId:      <domain id, any string>
```

Comment:
I don't recall exactly how these properties relate to the configurations specified in the  `rabbitmq.instance.jsonlist` property, but unless the latter is set none of the properties above matter. Please see #176 for a description of how confusing all of this is. As I write in that issue I think the configuration interface should be overhauled completely, but the least we can do is document the current interface.